pyresample/resampler.py

```diff
@@ -38,6 +47,138 @@ def hash_dict(the_dict, the_hash=None):
     return the_hash
 
 
+def add_xy_coords(data_arr, area, crs=None):
+    """Assign x/y coordinates to DataArray from provided area.
+
+    If 'x' and 'y' coordinates already exist then they will not be added.
+
+    Args:
+        data_arr (xarray.DataArray): data object to add x/y coordinates to
+        area (pyresample.geometry.AreaDefinition): area providing the
+            coordinate data.
+        crs (pyproj.crs.CRS or None): CRS providing additional information
+            about the area's coordinate reference system if available.
+            Requires pyproj 2.0+.
+
+    Returns (xarray.DataArray): Updated DataArray object
+
+    """
+    coords_exist = 'x' in data_arr.coords and 'y' in data_arr.coords
+    no_xy_dims = 'x' not in data_arr.dims or 'y' not in data_arr.dims
+    if coords_exist or no_xy_dims:
+        # no defined x and y dimensions
+        return data_arr
+
+    if hasattr(area, 'get_proj_vectors'):
```

Comment:
These 3 conditions could be checked together in a separate method that can be called directly here to know if we can just return `data_arr`

Reply:
Good catch.This was copied over from Satpy and I was just updating it to stop deepcode from complaining. I'll update this in a bit, but I'm not pleased with the double negation I ended up having to do to make the method name make sense.

Reply:
Done.